### PR TITLE
Fix ipa-client-automount install/uninstall with new install states

### PR DESCRIPTION
**Pull request by rcritten**
Issue 8384 introduced a new installation state for the statestore
to identify when client/server installation is completely finished
rather than relying on has_files().

The problem is that ipa-client-automount may be called during
ipa-client-install and since installation is not complete at that
point the automount install was failing with "IPA client not
configured".

Add a new state, 'automount', to designate that automount installation
is in process. If check_client_configuration() fails it checks to
see if [installation] automount is True. If so it continues with the
installation.

This also addresses an issue where the filestore and statestore are
shared between the client and automount installers but the client
wasn't refreshing state after automount completed. This resulted in
an incomplete state and index file of backed-up files which caused
files to not be restored on uninstall and the state file to be
orphaned.

Fixes: https://pagure.io/freeipa/issue/9487

Signed-off-by: Rob Crittenden <rcritten@redhat.com>

### CODE REVIEW

File: ipaclient/install/ipa_client_automount.py [hunk 1]
```diff
@@ -340,14 +340,17 @@ def configure_nfs(fstore, statestore, options):
...
 
 def configure_automount():
+    statestore = sysrestore.StateFile(paths.IPA_CLIENT_SYSRESTORE)
     try:
         check_client_configuration()
     except ScriptError as e:
-        print(e.msg)
-        sys.exit(e.rval)
+        # Are we being called from ipa-client-install?
+        automount = statestore.get_state('installation', 'automount')
```
Comment (tiran) [inline ipaclient/install/ipa_client_automount.py:348]: I like your approach with the additional state variable. The flow is a bit hard to follow, though. How about you move the check for the new state var one level up and skip the client configure script when the state is set?

```suggestion
    if not statestore.get_state('installation', 'automount'):
        # not called from ipa-client-install
        try:
            check_client_configuration()
        except ScriptError as e:
            print(e.msg)
            sys.exit(e.rval)
```

Comment (rcritten) [reply ipaclient/install/ipa_client_automount.py]: Yes, that is much simpler. Done.

**Comment by flo-renaud** [comment]
@rcritten thanks for the PR, works for me. Tested the following scenarios:
- [x] install client with automount parameter, uninstall
- [x] install client, configure automount, uninstall
- [x] install client, configure automount, unconfigure automount, uninstall

Not sure why the CI is re-running everything (with your temp commit I would expect only fedora-latest/temp_commit to be launched...) You can remove the temp commit and I will ack.

**Comment by flo-renaud** [comment]
master:

* e4420624ffed47c42b3bd0dfd580cd98f667e843 Fix ipa-client-automount install/uninstall with new install states
* ce811db6be532a9f258d7429234028975eb99f50 ipatests: Test client install/uninstall with automount enabled
* 54fb1173f9ab1025c12a77b3a5bf205afa8f63e2 ipa-client-automount: Don't use deprecated ipadiscovery.IPADiscovery
